app/models/concerns/has_many_common.rb

```diff
@@ -73,6 +73,9 @@ def has_many_names_for(association, options)
     #### belongs_to ####
     def belongs_to(*args)
       options = args.last.is_a?(Hash) ? args.last : {}
+      if options.key?(:name_accessor)
+        Foreman::Deprecation.deprecation_warning('3.6', 'Passing name_accessor to belongs_to is deprecated, use native options instead')
```

Comment:
can we define the methods and throw the deprecation warning there - in the usage of the methods?

Reply:
My reasoning is that I want to get rid of this `belongs_to` definition altogether. Ideally only Rails defines it. Though after https://github.com/theforeman/foreman/pull/9332 it needs a rebase anyway.